### PR TITLE
add Selection List component

### PR DESCRIPTION
**Pull request by DanielHodgeGH**

### CODE REVIEW

File: src/Nox.Ui.Blazor.Lib/Components/Generic/SelectList.razor.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+{
+
+    #region Declarations
+
+    [Parameter]
+    public List<SelectEntityModel> EntityList { get; set; } = new();
+
+    public string? CurrentIdList { get; set; }
+
+    public string? DisplaySelectedList { get; set; }
```
Comment (anna-naumova25) [inline src/Nox.Ui.Blazor.Lib/Components/Generic/SelectList.razor.cs:18]:
```suggestion
    public string? DisplaySelectedList => EntityList.Any() ? string.Join(", ", EntityList.Select(Entity => Entity.Name)) : string.Empty;
```

File: src/Nox.Ui.Blazor.Lib/Components/Generic/SelectList.razor.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
+﻿using Microsoft.AspNetCore.Components;
+using Microsoft.VisualBasic;
+using MudBlazor.Charts;
+using Nox.Ui.Blazor.Lib.Models;
+
+namespace Nox.Ui.Blazor.Lib.Components.Generic;
+
+public partial class SelectList : ComponentBase
```
Comment (anna-naumova25) [inline src/Nox.Ui.Blazor.Lib/Components/Generic/SelectList.razor.cs]: Maybe I'd rename it to SelectMany. @rochar what do you think?

**Review by anna-naumova25 — DISMISSED** [review]
Looks good, just one small comments that can be done in the next PR.

File: src/Nox.Ui.Blazor.Lib/Components/Generic/SelectList.razor [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+               FullWidth="true"
+               Clearable="true"
+               Required="@Required"
+               RequiredError="@ErrorRequiredMessage(Title)"
+               TextChanged="OnIdChanged"
+               Value="@CurrentIdList"
+               Text="@DisplaySelectedList">
+        @foreach (SelectEntityModel item in SelectionList)
+        {
+            if (!string.IsNullOrWhiteSpace(item.Id)
```
Comment (rochar) [inline src/Nox.Ui.Blazor.Lib/Components/Generic/SelectList.razor]: @DanielHodgeGH  this safe guard can hide or is hidding some issue in the domain

can not exist an entity without an Id or a text field, we should not support such cases

I would prefer to remove the safe guard and see if we need to fix anything in other place?

File: src/Nox.Ui.Blazor.Lib/Components/Generic/SelectList.razor.cs (hunk 2 above)
Comment (rochar) [reply src/Nox.Ui.Blazor.Lib/Components/Generic/SelectList.razor.cs]: maybe MultiSelect?

SelectMany is more about flatten a lists of lists

File: src/Nox.Ui.Blazor.Lib/Components/Generic/SelectList.razor.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        CurrentIdList = newValue;
+        IEnumerable<string> selectedIdList = new List<string>();
+        if (!string.IsNullOrWhiteSpace(CurrentIdList))
+        {
+            selectedIdList = CurrentIdList.Split(',').Select(Id => Id.Trim()).ToList();
+        }
+
+        EntityList = new();
+
+        if (selectedIdList.Any())
```
Comment (rochar) [inline src/Nox.Ui.Blazor.Lib/Components/Generic/SelectList.razor.cs]: this is unnecessary, the foreach does this implicitly?

File: src/Nox.Ui.Blazor.Lib/Components/Generic/SelectList.razor.cs [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        {
+            foreach (string currentId in selectedIdList)
+            {
+                SelectEntityModel? entity = SelectionList.Find(Entity => 
+                !string.IsNullOrWhiteSpace(Entity.Id)
```
Comment (rochar) [inline src/Nox.Ui.Blazor.Lib/Components/Generic/SelectList.razor.cs]: the same here, actually at the moment we are not adding an entity if the id is empty...

**Review by rochar — CHANGES_REQUESTED** [review]
looks good, a few comments to review

**Review by anna-naumova25 — APPROVED** [review]

**Review by rochar — APPROVED** [review]

**Pull request merged**